### PR TITLE
Added a few power ups

### PR DESCRIPTION
**Pull request by BenJacobs-dev**

### CODE REVIEW

File: entities/player/Player.gd [hunk 1]
```diff
@@ -38,17 +43,21 @@ func _physics_process(_delta):
 	_process_collisions()
 	move_and_slide()
 
+func damage_player(amount):
+	if is_invincible:
+		return
```
Comment (rgreenan55) [inline entities/player/Player.gd]: I think godot typically uses pass but I could be wrong

Comment (BenJacobs-dev) [reply entities/player/Player.gd]: nah, pass is just a no op, return ends the function early

File: entities/player/Player.gd [hunk 2]
```diff
@@ -38,17 +43,21 @@ func _physics_process(_delta):
 	_process_collisions()
 	move_and_slide()
 
+func damage_player(amount):
```
Comment (rgreenan55) [inline entities/player/Player.gd]: Okay this function looks good. I have it named handle_hit and it simply reduces the value by 1. And I created a custom setter function for health so we can move the code for is_dead there.

Or in _process we can just check every frame if health == 0 and then call die() function if so and handle any death related functionality there

Is it possible for you to combine this invincibility code with something like on_hit the player can't take more damage for 0.5 like you did invincibility?

Comment (BenJacobs-dev) [reply entities/player/Player.gd]: I'm not fully sure what you mean but I can make the invincibility code work with whatever function you make, once you merge it I'll figure it out

**Pull request merged**